sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -95,7 +95,8 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case Filter(condition, child) =>
       val (withSubquery, withoutSubquery) =
-        splitConjunctivePredicates(condition).partition(SubqueryExpression.hasInOrExistsSubquery)
+        splitConjunctivePredicates(condition)
+          .partition(SubqueryExpression.hasCorrelatedInOrExists)
```

Comment:
This change makes the rule `RewritePredicateSubquery`'s document out-of-date. It also might be named as `RewriteCorrelatedPredicateSubquery`?

Reply:
Agree, but need other reviewers' suggestion.

Reply:
+1 on `RewriteCorrelatedPredicateSubquery`